etna/packages/etna-py/src/mountetna/metis.py

```diff
@@ -504,27 +521,22 @@ def upload_parts(
                 )
 
                 unsent_zero_byte_file = False
-            except RequestException as e:
-                if remaining_attempts > 1:
-                    sleep(30)
-                    if e.response.status_code == 422:
-                        yield from self.upload_parts(
-                            upload, metis_uid, remaining_attempts - 1, True
-                        )
-                        return
-                    else:
-                        yield from self.upload_parts(
-                            upload, metis_uid, remaining_attempts - 1
-                        )
-                        return
-                raise e
+            except HTTPError as e:
+                if remaining_attempts > 1 and e.response.status_code == 422:
+                    yield from self.upload_parts(
+                        upload, metis_uid, remaining_attempts - 1, True
+                    )
+                    return
+                else:
+                    raise e
 
             yield upload
 
     def create_folder(
         self, project_name: str, bucket_name: str, folder_path: str
     ) -> FoldersAndFilesResponse:
-        response = self.session.post(
+        response = self._retry_request(
```

Comment:
Note:  Retry here is tricky, because it's not idempotent.  For instance, it's possible the first request DOES create the folder, but the connection breaks or there is a 500 due to some other subsequent logic bug (maybe just apache has an issue).  So then the client retries -- but this time it returns a invalid request exception because the folder exists from the previous request.  In this case, a lot of `create_folder` calls may throw false negative exceptions when all that is happening is retry logic is kicking in.

Ideally, we would make most metis requests idempotent (for instance, not throwing an exception if two identical requests are made).  But that may be a bit of work.